src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionValidator.java

```diff
@@ -16,153 +16,188 @@
+		String regex = ".*[\\s].*";
+		if (notificationMeta == null) {
+			throw new SubscriptionValidationException("Required field NotificationMeta has not been set");
+		} else if (Pattern.matches(regex, notificationMeta)) {
+			throw new SubscriptionValidationException("Wrong format of NotificationMeta: " + notificationMeta);
+		}
+	}
+
+	/**
+	 * Validation of notificationType parameter Throws
+	 * SubscriptionValidationException if validation of the parameter fails due to
+	 * wrong format of parameter.
+	 *
+	 * @param notificationType
+	 */
+	private void validateNotificationType(String notificationType) throws SubscriptionValidationException {
+		String regexMail = "[\\s]*MAIL[\\\\s]*";
+		String regexRestPost = "[\\s]*REST_POST[\\\\s]*";
+		if (notificationType == null) {
+			throw new SubscriptionValidationException("Required field NotificationType has not been set");
+		} else if (!(Pattern.matches(regexMail, notificationType)
+				|| Pattern.matches(regexRestPost, notificationType))) {
+			throw new SubscriptionValidationException("Wrong format of NotificationType: " + notificationType);
+		}
+	}
+
+	private void RestPostMediaType(String restPostMediaType) throws SubscriptionValidationException {
+		String regexApplication_JSON = "[\\s]*application/json[\\\\s]*";
+		String regexApplicationFormUrlEncoded = "[\\s]*application/x-www-form-urlencoded[\\\\s]*";
+		if (restPostMediaType == null) {
+			throw new SubscriptionValidationException("Required field RestPostMediaType has not been set");
+		} else if (!(Pattern.matches(regexApplication_JSON, restPostMediaType)
+				|| Pattern.matches(regexApplicationFormUrlEncoded, restPostMediaType))) {
+			throw new SubscriptionValidationException("Wrong format of RestPostMediaType: " + restPostMediaType);
+		}
+	}
+
+	/**
+	 * Validation of email address Throws SubscriptionValidationException if
+	 * validation of the parameter fails due to wrong format of parameter.
+	 *
+	 * @param email
+	 */
+	public void validateEmail(String email) throws SubscriptionValidationException {
+		final Pattern VALID_EMAIL_ADDRESS_REGEX = Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$",
+				Pattern.CASE_INSENSITIVE);
+		Matcher matcher = VALID_EMAIL_ADDRESS_REGEX.matcher(email);
+		if (!(matcher.matches())) {
+			throw new SubscriptionValidationException("Wrong email address: " + email);
+		}
+	}
+
+	public void validateWithSchema(Subscription subscription) throws SubscriptionValidationException {
+		LOGGER.debug("Validation of subscription " + subscription.getSubscriptionName() + " Started.");
+
+		ObjectMapper mapper = new ObjectMapper();
+		String subscriptionJson = null;
+		try {
+			subscriptionJson = mapper.writeValueAsString(subscription);
+		} catch (JsonProcessingException e) {
+			// TODO Auto-generated catch block
+			LOGGER.error("Failed to create object to json" + "\nError message: " + e.getMessage(), e);
```

Comment:
You should trow exception, since execution can't continue when subscriptionJson is null here.
You should add throw exception similar to this: "throw new SubscriptionValidationException("<failure message>" + e.getMessage());"

Reply:
done